collie/model/hybrid_pretrained_matrix_factorization.py

```diff
@@ -242,27 +308,83 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
             Predicted ratings or rankings
 
         """
-        if str(self.device) != str(self.item_metadata.device):
-            self._move_any_external_data_to_device()
 
-        metadata_output = self.item_metadata[items, :]
-        if self.metadata_layers is not None:
-            for metadata_nn_layer in self.metadata_layers:
-                metadata_output = self.dropout(
+        if self.item_metadata is not None:
+            if str(self.device) != str(self.item_metadata.device):
+                self._move_any_external_data_to_device()
+        if self.user_metadata is not None:
+            if str(self.device) != str(self.user_metadata.device):
```

Comment:
[Optional] Can we refactor this to be a single `if` statement so we only have to call `self._move_any_external_data_to_device()` once?

Reply:
👍